src/threadsync.cpp

```diff
@@ -202,6 +204,10 @@ ThreadSync::resetLocks()
   _checkpointThreadInitialized = false;
   _wrapperExecutionLockAcquiredByCkptThread = false;
   _threadCreationLockAcquiredByCkptThread = false;
+
+  if (resetPresuspendEventHookLock) {
```

Comment:
This seems to create a memory leak if it is executed more than once.  Shouldn't you destroy the mutex first, and then initialize a new mutex?